src/reporter/tests/test_version.py

```diff
@@ -4,8 +4,19 @@
 
 def test_version():
     version_url = "{}/version".format(QL_URL)
-    r = requests.get('{}'.format(version_url))
+    QL_version_url = version_url.replace("/v2","")
+    r = requests.get('{}'.format(QL_version_url))
```

Comment:
hi @chicco785 , Thank you for your suggestion. As per my understating editing these two test files- (such as test_version.py and test_health.py ) would be better instead of changing the QL_URL in all other test suites.

Here I have changed /v2/version to /version by removing /v2 from QL_URL in test_version.py.
If I remove /v2 from QL_URL then I will have to add /v2 in all the other APIs.
Please confirm the approach I should take.

Reply:
correct, i did a quick check, and i believe these are the lines you will need to change:
- https://github.com/smartsdk/ngsi-timeseries-api/blob/095244b290d84ae5f75edaccdb4c2b1f2f46f2df/src/reporter/tests/test_subscribe.py#L5
- https://github.com/smartsdk/ngsi-timeseries-api/blob/7b9366181807d62ec14600771b9ddb95a8eb1b6a/src/reporter/tests/test_delete.py#L7
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/geo_queries_fixture.py#L37
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_attribute_name_case.py#L60
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_attribute_name_case.py#L68
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_attribute_name_case.py#L75
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_multitenancy.py#L36
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_multitenancy.py#L48
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b33a1a5e4321f198ffbc65abe5a43d82ac2bd887/src/reporter/tests/test_integration.py#L13
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b33a1a5e4321f198ffbc65abe5a43d82ac2bd887/src/reporter/tests/test_integration.py#L44
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b33a1a5e4321f198ffbc65abe5a43d82ac2bd887/src/reporter/tests/test_integration.py#L68
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b33a1a5e4321f198ffbc65abe5a43d82ac2bd887/src/reporter/tests/test_integration.py#L103
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_entities_with_odd_chars.py#L29
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_entities_with_odd_chars.py#L36
- https://github.com/smartsdk/ngsi-timeseries-api/blob/a99e701b311868f26296c16ff730b8015ff31ef0/experiments/iotagent/subscribe_ql.py#L13
- https://github.com/smartsdk/ngsi-timeseries-api/blob/bfac4d351c9eb9b01ba006171d6ba00159feaa1f/src/reporter/tests/utils.py#L9
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_incomplete_entities.py#L15
- https://github.com/smartsdk/ngsi-timeseries-api/blob/a65c2038acba2a77b6016876bd07203c950f140f/src/reporter/tests/test_1T1E1A.py#L16
- https://github.com/smartsdk/ngsi-timeseries-api/blob/a65c2038acba2a77b6016876bd07203c950f140f/src/reporter/tests/test_1T1E1A.py#L384
- https://github.com/smartsdk/ngsi-timeseries-api/blob/b234a90ca0725bd869e38a8b2efeec047b04b7f7/src/reporter/tests/test_1TNE1A.py#L14
- https://github.com/smartsdk/ngsi-timeseries-api/blob/56cb44b5e34b7dc0cf6eadae4764e56f498ae2ef/src/reporter/tests/test_notify.py#L10
- https://github.com/smartsdk/ngsi-timeseries-api/blob/a65c2038acba2a77b6016876bd07203c950f140f/src/reporter/tests/test_1T1ENA.py#L17

Reply:
Hi @c0c0n3 @chicco785 ,

I have fixed all above files and also all test cases are running successfully but an error `404 error`  is returning on execution of following files: test_notify.py, test_multitenancy.py, test_integration.py

As per my analysis and code debugging of QL for above error, i found that entity is created successfully on Orion with entityid- `Room1` but the error is returning in GET request of entity.

Moreover, I had debugged the mentioned file- `/src/reporter/query_1T1ENA.py` further and found that in crate.py the schema is not able to create and showing, `Schema : "mtmyservice" Unknown error`. 

Please find below the attached screenshot for the same:

![image](https://user-images.githubusercontent.com/35693908/67272158-a253eb00-f4d9-11e9-9750-be3eb81b5a75.png)

The SQL query created during the test execution is: 
``` 
('select * from "mtmyservice"."etroom" where  entity_id in (\'Room1\') and  '
 "fiware_servicepath ~* '/($|/.*)'  ORDER BY time_index ASC limit 10000 offset "
 '0')
```

So, to verify the same, I had triggered the same request API `http://localhost:8668/v2/entities/Room1` on QL after creating the same entity in Orion as written in the test case(`Fiware-Service: myservice `and `Fiware-ServicePath: /` being also the same.). And the SQL query was also similar to the one created during test case execution of test_multitenancy.py 

```
('select * from "mtmyservice"."etroom1" where  entity_id in (\'Room1\') and  '
 "fiware_servicepath ~* '/($|/.*)'  ORDER BY time_index ASC limit 10000 offset "
 '0')
```

Thus, I found it very contrasting that when we send the same parameters- headers etc and secondly the SQL query being also the same, the behavior observed is quite different in both these scenarios. 

Could you please confirm us that where are we going wrong? I'm still working on this issue.

Reply:
Could you please confirm  @chicco785  @c0c0n3 .Thanks in advance

Reply:
@daminichopra we found a bug in the build process due to a change in the alpine base image that is used to packaged quantumleap, as first step, can you rebase the code on the master branch?

Reply:
@chicco785, I have updated the branch to master,now would request you to  please confirm with the understanding?

Reply:
@daminichopra I don't think we should change any of the test files we told you to change earlier---sorry about that, the code is tangled and sometimes it gets confusing for us too :-)

Since basically you're only changing the path of the health, version, and config endpoints, it makes sense to only update the corresponding tests. Changing the QL_URL would trigger way too many side effects. So I suggest we add a QL_BASE_URL instead and use that variable in those test cases. The idea is that QL_URL means "API base URL ie. host/v2" whereas QL_BASE_URL is just the host part---perhaps you can come up with a better variable name :-)
Anyhoo, I've just pushed a branch w/ my suggestions for a baseline, here's a diff w/ master:

https://github.com/smartsdk/ngsi-timeseries-api/compare/pr-261-suggestion

As you can see it's basically the same as what you've done except I introduced that QL_BASE_URL to avoid changing tests that have nothing to do w/ your change---e.g. in principle there should be no need to update reporter query tests since they were pointing to `host/v2` before your change and should keep on using `host/v2` after your change too.

Note that I sneakily zapped `config_url` and `health_url` from the JSON returned by `list_of_api`. You did that in the test case, so I assume that's what you wanted to happen in `list_of_api` too.

Now for the fun part. There's a test that's still failing, namely `src/tests/test_bc.py`. I didn't have time to figure out why, but would greatly appreciate if you could look into that :-)

Thank you sooo much for the time and effort you've put in to make QL better, we really value your contributions!

Reply:
> Now for the fun part...There's a test that's still failing, namely `src/tests/test_bc.py`

I figured out what the problem was and pushed a solution to the "suggestion" branch:

https://github.com/smartsdk/ngsi-timeseries-api/compare/pr-261-suggestion

Basically there was another hairy knot to undo. The backward compatibility test first loads some data in an old QL image (version `0.5`) and then checks the latest QL can still digest it. But before shoveling data in, the test case runs a sanity check to see if we can get to QL at all by hitting the `version` endpoint. Now that endpoint will be at `http://<host>:<port>/v2/version` for QL `0.5` whereas it'll have to be at `http://<host>:<port>/version` for the code brought in by this PR.